### PR TITLE
[4.0] Javascript clean up

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- remove or move code out of core.js
- Rename global functions to Joomla.functionName (for testing and because it's not nice to pollute to have global functions)
- fix the multi select console error

### Testing Instructions
Apply patch and test any list view (e.g. articles that ALL the links, buttons work as expected)
Try the sendmail in Global Configuration
And also in the global configuration [System Settings] try the button refresh for the Help Server

### CODE REVIEW

**Comment by C-Lodder** [comment]
Working so far:

- [x] Send mail
- [x] Help server refresh
- [x] List view toolbar buttons
- [x] List views multiselect
- [ ] User field

**Comment by rjcf18** [comment]
I have tested this item :red_circle: unsuccessfully on a4a086a31872c5d2e0a4e5de9c117f2e4e1486e6<br /><br />So in the articles view in com_content when I click on the publish/unpublish and feature toggle buttons in the status column an error occurs. 
deepinscreenshot20170425211828
deepinscreenshot20170425211849

Also when I order by status for instance and when hovering on other buttons like the ones on the status column another error occurs. Try hovering around and clicking to replicate this.
deepinscreenshot20170425213305

Regarding the rest of the buttons and links I've tried several of them and all seems to be working fine.
deepinscreenshot20170425211928

The rest (send mail and refresh help server) I'm not sure if it is working fine, not sure if the behaviour I've witnessed is the expected one.
Regarding the send mail part no errors in the console happened but an error message popped up, not sure if that was the expected behaviour:
deepinscreenshot20170425212335

And finally regarding the help server refresh no JS errors spotted and nothing happened on click, also not sure if this was the intended behaviour. 
deepinscreenshot20170425212629
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15533">issues.joomla.org/tracker/joomla-cms/15533</a>.</sub>

**Comment by dgt41** [comment]
@rjcf18 thanks!
- So the first 2 are bugs but it should be fixed now.
- tooltips error is due to bootstrap (is all over the backend not related to this PR)
- Send mail is correctly showing you the error message
- Help refresh is just refreshing the dropdown

**Comment by rjcf18** [comment]
Ah ok nice! :) 
The last two I was a bit unsure. I'll test again to check the other two bugs.

**Comment by rjcf18** [comment]
I have tested this item :red_circle: unsuccessfully on 3ba1f01e849ebee39648bba57d252871b26d8fe9<br /><br />There is still an error, a different one this time, in those same two buttons from the status column:
deepinscreenshot20170425234503
deepinscreenshot20170425234242<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15533">issues.joomla.org/tracker/joomla-cms/15533</a>.</sub>

**Comment by dgt41** [comment]
@rjcf18 sorry my bad, wrong copy paste...

**Comment by rjcf18** [comment]
I have tested this item :white_check_mark: successfully on beabb764cd4b5cb1ed4552b8efb24f62bb24924d<br /><br />Problem has been solved, the two buttons on the status column are now working and no errors were spotted on console.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15533">issues.joomla.org/tracker/joomla-cms/15533</a>.</sub>

**Comment by ciar4n** [comment]
I have tested this item :white_check_mark: successfully on beabb764cd4b5cb1ed4552b8efb24f62bb24924d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15533">issues.joomla.org/tracker/joomla-cms/15533</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Comment by wilsonge** [comment]
Can we deprecate the core.js and fix conflicts please?

**Comment by dgt41** [comment]
> Can we deprecate the core.js

The idea is to clean up the bloated code from core.js, rename that file to joomla.js or system.js and also introduce JHtml::_('behavior.system') (or something like that. So in short, core.js **will be the same as 3.7** 
The main thing with the javascript is to establish a strategy to create the final files

File: media/system/js/multiselect.js [hunk 1]
```diff
@@ -47,44 +47,195 @@ Joomla.JMultiSelect = function(table) {
 	initialize(table);
 };
```
Comment (Fedik) [inline media/system/js/multiselect.js]: whole file need to be wrapped in to an anonymous function with use `'use strict';` mode, 
as it in `core.js`

**Comment by dgt41** [comment]
I will redo this one later on..

**Pull request closed without merging**